### PR TITLE
Merge latest improvements into production

### PR DESCRIPTION
**Pull request by tcouch**
### Include unlinked fragments in @mentions  
* Issue: #295 
* PR: #305
Unlinked fragments can now be added via @ mentions. This works in a similar fashion to anonymous fragments search in that a keyword consisting of the letter "u" will find all unlinked fragments, and "u<int>" will find all unlinked fragments whose id starts with that number; e.g. "u33" will find Unlinked 338 and Unlinked 339.

### Work fragments ordered by book
* Issue: #160 
* PR: #304 
Material listed on a work's detail page is now organised by book. Anchor links have been added from the books listed at the top to the area of the page containing material for those books.
#### Examples:
* https://www.ucl.ac.uk/history/frrant-preprod/work/37/
* https://www.ucl.ac.uk/history/frrant-preprod/work/27/

### Fix display of apposita for unlinked fragments
* Issue: #289 
* PR: #303 
#### Examples:
* https://www.ucl.ac.uk/history/frrant-preprod/anonymous/53/

### Ignore angle brackets in search queries
* PR: #302
#### Examples:
* https://www.ucl.ac.uk/history/frrant-preprod/search/?q=findthem&what=all 

### Refuse connections to unexpected locations
* PR: #307 
Technical fix to the webserver refuses connection to unrecognised locations

### CODE REVIEW

**Comment by tcouch** [comment]
Hi @rmamarshall. The changes listed above are now ready to be added to production. Could you review these on the [development server](https://www.ucl.ac.uk/history/frrant-preprod/) and let me know if you're happy for us to add them please?

**Comment by rmamarshall** [comment]
Hi @tcouch. The five changes look really good and everything works as expected, except "Fix display of apposita for unlinked fragments". I turned a fragment with apposita into an unlinked fragment, and the apposita were retained. However, when I next turned the new unlinked fragment into an anonymous fragment, the links disappeared. As Anonymous fragments can also have apposita, this may be an issue. A similar issue was encountered when I turned an unlinked fragment into a non-anonymous fragment: all apposita links disappeared. I think we may have discussed this issue before - apologies if I have got the wrong end of the stick and the above fix is for a different feature!

**Comment by tcouch** [comment]
Hi @rmamarshall. Thanks for the feedback. Can I just clarify a couple of things?

I was under the impression that Anonymous Fragments can be apposita to Fragments, but they can't be apposita to each other. I presumed this was a deliberate decision as that's how the database has been designed. So, when converting an unlinked fragment with apposita into an anonymous fragment, it would have to lose its links to other anonymous fragments.

When you say "when I turned an unlinked fragment into a non-anonymous fragment...", did you mean something else? I was confused because, unlinked fragments **are** non-anonymous fragments.

**Comment by rmamarshall** [comment]
Hi @tcouch. Apologies, I did indeed have the wrong end of the stick: Anonymous fragments should not have apposita. The other issue I described arose from turning an unlinked fragment into an anonymous one, then turning the anonymous fragment into a non-anonymous one: because of the intermediate step, the apposita were lost, but I don't see why this should happen in actual usage (unliked fragments will simply be linked to authors / works directly). Did we discuss a possible warning pop-up at some point, alerting users to the fact that making a fragment with apposita into an anonymous fragment will remove the apposita links?

File: src/config/settings/development.py [hunk 1]
```diff
@@ -97,13 +97,14 @@ def monkey_patch():
 # https://docs.djangoproject.com/en/dev/ref/settings/#default-from-email
 DEFAULT_FROM_EMAIL = env(
     "DJANGO_DEFAULT_FROM_EMAIL",
-    default="Republican Antiquarians Research Database <noreply@example.com>",
+    default="Republican Antiquarians Research Database (dev) <noreply@example.com>",
```
Comment (raquelalegre) [inline src/config/settings/development.py:100]: It shouldn't say "(dev)" in the production version, right? This line shouldn't be merged.

File: src/config/settings/development.py [hunk 2]
```diff
@@ -97,13 +97,14 @@ def monkey_patch():
...
 )
 # https://docs.djangoproject.com/en/dev/ref/settings/#server-email
 SERVER_EMAIL = env("DJANGO_SERVER_EMAIL", default=DEFAULT_FROM_EMAIL)
 # https://docs.djangoproject.com/en/dev/ref/settings/#email-subject-prefix
 EMAIL_SUBJECT_PREFIX = env(
-    "DJANGO_EMAIL_SUBJECT_PREFIX", default="[Republican Antiquarians Research Database]"
+    "DJANGO_EMAIL_SUBJECT_PREFIX",
+    default="[Republican Antiquarians Research Database (dev)]",
```
Comment (raquelalegre) [inline src/config/settings/development.py:107]: Same as earlier comment.

**Review by raquelalegre — APPROVED** [review]
Looks great. Just made 2 small comments.

File: src/config/settings/development.py (hunk 1 above)
Comment (tcouch) [reply src/config/settings/development.py:100]: This settings file is only used by the development server. The production server uses src/config/settings/production.py which has these variables set as suggested.

File: src/config/settings/development.py (hunk 2 above)
Comment (tcouch) [reply src/config/settings/development.py:107]: See my reply to your comment above.

**Comment by tcouch** [comment]
Hi @rmamarshall I've updated the warning message to mention that apposita links will be lost and added that to this pull request. Are you happy for me to go ahead and put this into production now? If so you can click to approve these changes, or just reply to this message.

File: src/config/settings/development.py (hunk 1 above)
Comment (raquelalegre) [reply src/config/settings/development.py:100]: Ah yes :) Silly me!

**Comment by rmamarshall** [comment]
Hi @tcouch Happy New Year! I'm not sure I understand - is this ready to trial in the pre-production server, or will it simply be added to the production version?

**Comment by tcouch** [comment]
Hi @rmamarshall, yes you can test this out on the pre-production server now. If you're happy with it, let me know and I'll put it into production.

**Comment by tcouch** [comment]
Hi @rmamarshall, can you confirm for me if you're happy for these changes to be merged into production now?

**Comment by rmamarshall** [comment]
Hi @tcouch I've just checked the new feature in the pre-production server and it seems to work perfectly; please merge it into production.

**Pull request merged**